### ISSUE TITLE
Replace Travis with GitHub Actions

### ISSUE THREAD

**Issue by nichtich** [LibreCat/Catmandu#385]
Travis does not work anymore. The gh-actions branch contains a first try to use GitHub Actions instead but [it fails on some Perl versions](https://github.com/LibreCat/Catmandu/actions/runs/1520776790).

**Comment by zmughal** [comment]
The failure is because `App::Cmd` needs Perl ≥ v5.20.0:

```
Entering App-Cmd-0.334
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.78 ... Yes (7.62)
Running Makefile.PL
Configuring App-Cmd-0.334 ... Perl v5.20.0 required--this is only v5.14.4, stopped at Makefile.PL line 5.
```

**Comment by jorol** [comment]
Thanks for the hint. I think another problem is https://github.com/perl4lib/marc-perl/issues/14.

**Comment by vpeil** [comment]
> Thanks for the hint. I think another problem is [perl4lib/marc-perl#14](https://github.com/perl4lib/marc-perl/issues/14).

But why? I can't see this dependency in cpanfile.

**Comment by vpeil** [comment]
@zmughal you are right. Here's the proof: https://github.com/LibreCat/Catmandu/runs/4367128386?check_suite_focus=true#step:6:675

I think there's no other way than requiring perl 5.20 for catmandu if we need `App::Cmd` (and perl 5.20 is from 2014)

**Comment by nichtich** [comment]
I don't see [any good reason](https://metacpan.org/dist/App-Cmd/changes) why App::Cmd requires Perl 5.20 since 2021. Is it possible to use an older version of App:Cmd, such as 0.330?

**Comment by jorol** [comment]
@vpeil Sorry for the confusion. We encountered the problem first when building Catmandu::PICA, where Catmandu::SRU also failed to build.

**Comment by zmughal** [comment]
> I don't see [any good reason](https://metacpan.org/dist/App-Cmd/changes) why App::Cmd requires Perl 5.20 since 2021.

The reason `App::Cmd` bumped up the Perl requirement was because the code started using postderef.

>  Is it possible to use an older version of App:Cmd, such as 0.330?

Yes, but this would need to specified to `cpanm` as:

```shell
cpanm 'Getopt::Long::Descriptive~<0.105' 'App::Cmd~<0.332'  Catmandu
```

which can be added to the CI for testing against older Perls only. That way if somebody does have a need for running on older Perls for now, they have a way to do that which has been tested. The drawback is that it won't benefit from any newer bug-fixes in the future.

**Comment by zmughal** [comment]
> > Is it possible to use an older version of App:Cmd, such as 0.330?
> 
> [...] can be added to the CI for testing against older Perls only.

I have now done this on the [`zmughal:gh-actions`](https://github.com/LibreCat/Catmandu/compare/gh-actions...zmughal:gh-actions?expand=1) branch.

All that remains is to run Perl::Tidy on the files so that the release tests
pass.

**Comment by nics** [comment]
resolved by #386
